collector/src/descriptor/kind/bridge_network_status.rs

```diff
@@ -0,0 +1,199 @@
+use derive_builder;
+use std::net::SocketAddrV6;
+use std::{
+    vec,
+    collections::HashMap,
+    net::{Ipv4Addr, SocketAddrV4},
+};
+
+use chrono::{DateTime, Utc};
+use derive_builder::Builder;
+
+use super::utils::*;
+use crate::error::{Error, ErrorKind};
+
+#[derive(Debug)]
+pub struct Header {
+    pub published_timestamp: DateTime<Utc>,
+    pub flags: HashMap<String, String>,
+    pub fingerprint: String,
+}
+
+impl Header {
+    pub fn parse(input: &str) -> Result<Self, Error> {
+        use crate::descriptor::nom_combinators::*;
+        let mut desc = descriptor_lines(input)?;
+        Ok(extract_desc! {
+        desc => Header rest {
+            uniq("published") [day, hour] => {
+                published_timestamp: date(&format!("{} {}", day, hour))?.1,
+            },
+           uniq("flag-thresholds") [] => {
+                // TODO should reject when split_once fail
+                flags: rest.iter()
+                            .filter_map(|v| v.split_once('='))
+                            .map(|(k,v)| (k.to_owned(), v.to_owned()))
+                            .collect(),
```

Comment:
you could map to a result, use something like [`Itertools.html::map_ok`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.map_ok), and collect to a `Result<Vec<_>,_>`